### PR TITLE
Fewer calls

### PR DESCRIPTION
**Pull request by deanajackson**
We would like to merge my original "fewer_calls" branch, that gave starting point variable for data for the visuals into Sienna's branch"component-Streak-Tracker" which has started working on said visuals. Lets at least 2/3 if not 3/3 take a look at this PR and see how we can merge it or make quick improvements. It's still a development branch so it doesn't have to be squeaky clean yet.

### CODE REVIEW

File: src/components/Dashboard.jsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  function createWeeksOnListArray(data) {
+    let weeksOnList = [];
+    let weeksByCat = [];
+    let temp = [];
+    data.forEach((list) => {
+      list.books.forEach((book) => {
+        weeksOnList.push(book.weeks_on_list);
+        temp.push(book.weeks_on_list);
+      });
+      weeksByCat.push(temp);
```
Comment (sieday) [inline src/components/Dashboard.jsx:69]: To make sure that these arrays stay connected to the correct list value, maybe this should be a dictionary instead. Use the list title as the key and the associated array as the value itself?

Comment (sieday) [reply src/components/Dashboard.jsx:69]: This could be helpful for any other 2d arrays based on list as well, such as the publishers section.

**Review by sieday — COMMENTED** [review]
I added a comment about the possibility of a dictionary for a couple sections, but that is the only big things I see. Thank you for your work on this! :)

**Comment by deanajackson** [comment]
I vote we merge the pull request and implement the dictionary feature in the next branch. I created the category array and the other arrays to make an "associative array" or simply an object to store key-value pairs or the .map() method? I had read this article https://flexiple.com/javascript/key-value-javascript which made me feel confident we could do this afterwards

**Pull request merged**